lib/quill_delta.dart

```diff
@@ -463,6 +486,59 @@ class Delta {
     return result;
   }
 
+  /// invert this delta base on baseDelta
+  Delta invert(Delta base) {
+    Delta inverted = new Delta();
+
+    if (base.isEmpty) return inverted;
+    int baseIndex = 0;
+
+    for (Operation op in _operations) {
+      if (op.isInsert) {
+        inverted.delete(op.length);
+      } else if (op.isRetain && op.attributes == null) {
+        inverted.retain(op.length, null);
+        baseIndex += op.length;
+      } else if (op.isDelete || (op.isRetain && op.attributes != null)) {
+        var length = op.length;
+        Delta sliceDelta = base.slice(baseIndex, baseIndex + length);
+        sliceDelta.toList().forEach((baseOp) {
+          if (op.isDelete) {
+            inverted.push(baseOp);
+          } else if (op.isRetain && op.attributes != null) {
+            var invertAttr = invertAttributes(op.attributes, baseOp.attributes);
+            assert(invertAttr.isNotEmpty);
+            inverted.retain(baseOp.length, invertAttr);
+          }
+        });
+        baseIndex += length;
+      }
+    }
+    inverted.trim();
+    return inverted;
+  }
+
+  /// get specific delta from start to end
+  Delta slice(int start, int end) {
+    var ops = [];
+    int index = 0;
+    var opIterator = new DeltaIterator(this);
+
+    while (index < end && opIterator.hasNext) {
+      Operation op;
+      if (index < start) {
+        op = opIterator.next(start - index);
+      } else {
+        op = opIterator.next(end - index);
+        ops.add(op);
+      }
+      index += op.length;
+    }
+    Delta newDelta = new Delta();
+    ops.forEach((op) => newDelta._operations.add(op));
+    return newDelta;
+  }
+
```

Comment:
Could you also add tests for `slice` from here: https://github.com/quilljs/delta/blob/master/test/delta/helpers.js#L190-L251